scapy/layers/dns.py

```diff
@@ -1046,6 +1051,55 @@ def pre_dissect(self, s):
 bind_layers(TCP, DNS, dport=53)
 bind_layers(TCP, DNS, sport=53)
 
+# Nameserver config
+conf.nameservers = read_nameservers()
+_dns_cache = conf.netcache.new_cache("dns_cache", 300)
+
+
+@conf.commands.register
+def dns_resolve(qname, qtype="A", verbose=1, **kwargs):
+    """
+    Perform a simple DNS resolution using conf.nameservers with caching
+    """
+    answer = _dns_cache.get("_".join([qname, qtype]))
+    if answer:
+        return answer
+
+    kwargs.setdefault("timeout", 5)
+    kwargs.setdefault("verbose", 0)
+    for nameserver in conf.nameservers:
+        try:
+            sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+            sock.connect((nameserver, 53))
+            sock = StreamSocket(sock, DNS)
+            res = sock.sr1(
+                DNS(qd=[DNSQR(qname=qname, qtype=qtype)]),
```

Comment:
We need to set the `DNS.id` field to a random value. As Scapy always uses 0 (`DNS(raw(DNS())).id == DNS(raw(DNS())).id == 0`), that could lead to some attacks.

Reply:
Good call !